Tiltfile.dev

```diff
@@ -1,6 +1,6 @@
 custom_build(
   ref = '169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/chs-delta-api',
-  command = 'DOCKER_BUILDKIT=0 docker build --build-arg SSH_PRIVATE_KEY="$(cat ~/.ssh/id_rsa)" --build-arg SSH_PRIVATE_KEY_PASSPHRASE --tag $EXPECTED_REF .',
+  command = 'DOCKER_BUILDKIT=0 docker build --build-arg SSH_PRIVATE_KEY="$(ssh_key_path="$(ssh -G github.com | grep -e \'^identityfile.*\' | head -n1 | sed \'s|^identityfile \\(.*\\)|\\1|\')"; if [ -z "${ssh_key_path}" ]; then echo "Could not find ssh key path for github.com">&2; false; elif [ -f "${ssh_key_path}" ]; then  cat "${ssh_key_path}";  else echo "Could not find ssh key for github at ${ssh_key_path}" >&2; false; fi)" --build-arg SSH_PRIVATE_KEY_PASSPHRASE --tag $EXPECTED_REF .',
```

Comment:
Looks fine, but perhaps worth a comment explaining the background issue and why this is needed. The long, single line of code is pretty unreadable.

Reply:
OK no worries. It's because the tilt up scripts assume that you are using a key called id_rsa but the dev-env-setup does not account for this, so you have to set up a github key manually otherwise the docker build fails. It also requires that your local ~/.ssh/config is set up slightly differently:

#github_start
Host github.com
  # ProxyCommand /usr/local/homebrew/bin/corkscrew websenseproxy.internal.ch 8080 %h %p
  Hostname ssh.github.com
  Port 443
  AddKeysToAgent yes
  IdentityFile /Users/someuser/.ssh/github_key
#github_end
#global_start
Host *
  ServerAliveInterval 60
  ServerAliveCountMax 10
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile /Users/someuser/.ssh/id_rsa
#global_end

The key point here is that the global config (denoted not by the comment #global_start but by the Host * bit) must appear AFTER the github config, otherwise the id_rsa key will be offered up again.